### PR TITLE
Data issues remediation task

### PR DESCRIPTION
**Pull request by yabirgb**
Part of https://github.com/rotki/rotki/issues/12289

Add background diagnostics for negative-balance data issues. The first strategy checks whether redecoding customized EVM transactions with the current decoders would change the affected balance, without replacing the user's saved events or customizations.

- Add the task interface and daily scheduling for data issue remediation, gated by the accounting update feature. Wait until historical balance processing has completed, track the last remediation run, and prevent scheduling it alongside history fetching, periodic historical balance processing, or another remediation run.
- Add virtual decoding of customized transactions using a discarded write buffer. Compare the ordered balance effects of saved events with freshly decoded events for the issue's account, asset, location, and protocol, respecting the ETH/ETH2 balance setting. Preserve saved history events and transaction decoded markers.
- Scope candidate transactions to the issue's account and time range using transaction-account associations, so customizations that change an event's asset or account label are still considered. Reuse transaction previews across issues within a run.
- Separate Ethereum produced-block fallback reward persistence from decoding through a post-decoding hook. Normal decoding still persists these staking rewards, while virtual decoding avoids that persistence step.
- Add strict error propagation for virtual decoding through input, address, event, transfer-enrichment, and post-decoding rules. A decoder failure produces a failed diagnostic instead of comparing partial results; regular decoding retains its existing error handling.
- Persist remediation attempts and show their outcome in the data issue timeline: redecoding would change the balance, would not change it, or failed. Record transaction counts and failure reasons. Issues pass through the auto-remediating state and remain unresolved for user review; the diagnostic does not automatically repair the balance.
- Add a debug-only `data_issue_remediation` option to the task-trigger API for on-demand runs. It bypasses the daily cooldown while preserving the feature gate and history-processing prerequisites.
- Add scheduler, API, decoder isolation/error propagation, account-scoping, persistence, and frontend timeline tests. Include a real flow test that creates a negative balance from a customized spend, runs scheduled remediation, and checks that the diagnostic is recorded without changing saved events.

The detailed before-and-after event comparison UI is in the stacked follow-up: https://github.com/yabirgb/rotki/pull/6.

### CODE REVIEW

File: rotkehlchen/tasks/data_issues.py [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+    from rotkehlchen.db.dbhandler import DBHandler
+    from rotkehlchen.history.events.structures.evm_event import EvmEvent
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+REDECODE_CUSTOMIZED_TRANSACTIONS: Final = 'redecode_customized_transactions'
+
+type BucketEffect = tuple[TimestampMS, int, EventDirection, str]
+type PreviewCache = dict[tuple[EVM_CHAIN_IDS_WITH_TRANSACTIONS_TYPE, EVMTxHash], list[EvmEvent]]
```
Comment (LefterisJP) [inline rotkehlchen/tasks/data_issues.py:51]:
```suggestion
  type BucketEffect = tuple[TimestampMS, EventDirection, FVal]
  ```
  Companion to the comment below on `_get_bucket_effects`: the raw sequence index should not be part of the compared tuple, and comparing `FVal` directly avoids depending on string formatting. Needs `from rotkehlchen.fval import FVal`.

File: rotkehlchen/tasks/data_issues.py [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+            event=event,
+            treat_eth2_as_eth=treat_eth2_as_eth,
+        ):
+            if event_bucket == bucket:
+                effects.append((
+                    event.timestamp,
+                    event.sequence_index,
+                    direction,
+                    str(event.amount),
+                ))
```
Comment (LefterisJP) [inline rotkehlchen/tasks/data_issues.py]:
```suggestion
                  effects.append((event.timestamp, direction, event.amount))
  ```
  Including `event.sequence_index` in the compared tuple produces false `redecoding_would_change_balance` results. Customized transactions keep the indices the decoder assigned when they were first decoded, and decoders reassign indices between releases (that is also why  
  `CUSTOMIZED_EVENT_ORIGINAL_SEQ_IDX` and the collision relocation in `_write_new_tx_events_to_the_db` exist). I verified it: a preview event with the same amount, direction and timestamp as the saved event but sequence index 7 instead of 0 is reported as
  `redecoding_would_change_balance` with `changed_transaction_count=1`. The UI would then tell the user to review a customization that changes nothing.

  The sort by `sequence_index` above already gives us the order, which is what matters for balance tracking (a spend before a receive inside one transaction can still go negative). The index value itself carries no balance information.

File: rotkehlchen/tasks/data_issues.py [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+        if len(customized_group_identifiers) == 0:
+            return {}
+
+        transaction_events = dbevents.get_history_events_internal(
+            cursor=cursor,
+            filter_query=EvmEventFilterQuery.make(
+                location=location,
+                group_identifiers=list(customized_group_identifiers),
+            ),
+        )
```
Comment (LefterisJP) [inline rotkehlchen/tasks/data_issues.py]: This runs once per open negative-balance issue per day, and issues without customized transactions stay `open`, so they are rescanned every day. As written it deserializes every customized event of the location and then every transaction of the account up to `ts_end` as 
  full `EvmTransaction` objects (input data included, plus the LEFT JOIN on authorizations), only to build a set of hashes.

  On a synthetic DB with 20k transactions for the account, 50 customized and 3k plain events, the current lookup takes 108.7 ms; a single join that returns the customized group identifiers directly takes 1.3 ms and materializes nothing but the ids. All the indexes it needs
  already exist (`idx_evmtx_address_mappings_address`, `idx_chain_events_info_tx_ref`, the `history_events_mappings` PK, `UNIQUE(tx_hash, chain_id)`).

  ```suggestion
      with database.conn.read_ctx() as cursor:
          customized_group_identifiers = [row[0] for row in cursor.execute(
              'SELECT DISTINCT H.group_identifier FROM history_events_mappings M '
              'JOIN history_events H ON H.identifier = M.parent_identifier '
              'JOIN chain_events_info C ON C.identifier = H.identifier '
              'JOIN evm_transactions T ON T.tx_hash = C.tx_ref AND T.chain_id = ? '
              'JOIN evmtx_address_mappings A ON A.tx_id = T.identifier AND A.address = ? '
              'WHERE M.name = ? AND M.value = ? AND H.location = ? AND T.timestamp <= ?',
              (
                  chain_id.serialize_for_db(),
                  issue.location_label,
                  HISTORY_MAPPING_KEY_STATE,
                  HistoryMappingState.CUSTOMIZED.serialize_for_db(),
                  location.serialize_for_db(),
                  ts_ms_to_sec(TimestampMS(issue.ts_end)),
              ),
          )]
          if len(customized_group_identifiers) == 0:
              return {}

          transaction_events = DBHistoryEvents(database).get_history_events_internal(
              cursor=cursor,
              filter_query=EvmEventFilterQuery.make(
                  location=location,
                  group_identifiers=customized_group_identifiers,
              ),
          )
  ```
  `HISTORY_MAPPING_KEY_STATE` comes from `rotkehlchen.db.constants`; `EvmAccount`, `string_to_evm_address`, `EvmTransactionsFilterQuery` and `Timestamp` become unused in this module. I checked the join returns the same transaction set as the current two-step lookup, and   
  all existing tests in `test_data_issue_remediation.py` exercise this path.

File: rotkehlchen/tasks/data_issues.py [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+    )
+    if len(transactions := _get_customized_transactions_for_issue(
+        database=database,
+        issue=issue,
+        chain_id=chain_id,
+        location=location,
+    )) == 0:
+        return
+
+    issues_manager.update_state(issue.id, IssueState.AUTO_REMEDIATING)
```
Comment (LefterisJP) [inline rotkehlchen/tasks/data_issues.py]:
```suggestion                                                                                                                         
      try:                                                          
          issues_manager.update_state(issue.id, IssueState.AUTO_REMEDIATING)
      except InputError:  # dismissed or resolved by the user after it was listed
          return                                                                                 
  ```
  This call is outside the `try` below. If the user dismisses or manually resolves the issue between `list_issues()` and here (a run with many previews can take a while), `_validate_state_transition` raises `InputError`, the whole task aborts with
  `exception_is_error=True`, the remaining issues are skipped, `LAST_DATA_ISSUE_REMEDIATION_TS` is not written, and the scheduler respawns the task on the next 10 s tick. It self-heals because the issue is no longer listed, but with an error log and a full repeated pass.
  `InputError` is in `rotkehlchen.errors.misc`.

File: rotkehlchen/tasks/data_issues.py [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+                )
+                preview_cache[chain_id, tx_hash] = preview_events
+            if _get_bucket_effects(saved_events, bucket, treat_eth2_as_eth) != _get_bucket_effects(
+                    preview_events,
+                    bucket,
+                    treat_eth2_as_eth,
+            ):
+                changed_transaction_count += 1
+            checkpoint()
+    except Exception as e:
```
Comment (LefterisJP) [inline rotkehlchen/tasks/data_issues.py]: `except Exception` is what we avoid in the backend. In strict mode the preview raises exactly the decoder's `possible_decoding_exceptions` (decode_safely re-raises only those) plus the `RuntimeError` from `_preview_transaction`, so the set is known. Something like:

  ```python                                                        
      except (RuntimeError, *chains_aggregator.get_evm_manager(chain_id).transactions_decoder.possible_decoding_exceptions) as e:  # noqa: E501                                                                                                                                  
  ```                                                                                                                                   
  or bind `decoder = chains_aggregator.get_evm_manager(chain_id).transactions_decoder` once at the top of `_check_issue` and pass it into `_preview_transaction`, which then no longer needs `chains_aggregator`. Anything outside that set is a real bug and should surface
  through `exception_is_error=True` instead of being recorded as a diagnostic outcome.

File: rotkehlchen/tasks/data_issues.py [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+    Saved events are never removed or replaced. Each applicable issue receives a diagnostic
+    timeline entry and remains unresolved for the user to review.
+    """
+    issues_manager = DataIssuesManager(database)
+    treat_eth2_as_eth = CachedSettings().get_entry('treat_eth2_as_eth') is True
+    preview_cache: PreviewCache = {}
+    for issue in issues_manager.list_issues(DataIssueFilters(
+        kind=IssueKind.NEGATIVE_BALANCE,
+        state=IssueState.OPEN,
+    )):
```
Comment (LefterisJP) [inline rotkehlchen/tasks/data_issues.py]: Once an issue has one attempt it is `unresolved` and this filter never sees it again. In strict mode any `RemoteError` (for example a token lookup during transfer decoding) or a missing receipt becomes `redecoding_failed`, so a transient network error permanently
  disables the diagnostic for that issue. There is no manual path either: `retry_auto_remediation` only accepts `rebasing_token`. The flow test encodes this as a strict xfail ("Step 5"), so I would rather land it here:
                                                                    
  ```python                                                         
      for issue in issues_manager.list_issues(DataIssueFilters(kind=IssueKind.NEGATIVE_BALANCE)):                                       
          if issue.state != IssueState.OPEN and not _last_attempt_failed(issue):
              continue                                                                                                                                                                                                                                                           
  ```                                                                                                                                                                                                                                                                            
  with                                                              

  ```python                                                                                                                             
  def _last_attempt_failed(issue: DataIssue) -> bool:
      """A failed comparison (missing receipt, network error) is retried on the next run."""                                                                                                                                                                                     
      return (       
          issue.state == IssueState.UNRESOLVED and                                                                                                                                                                                                                               
          len(issue.auto_remediation_attempts) != 0 and                                                                                                                                                                                                                          
          issue.auto_remediation_attempts[-1].get('strategy') == REDECODE_CUSTOMIZED_TRANSACTIONS and
          issue.auto_remediation_attempts[-1].get('result') == 'redecoding_failed'                                                                                                                                                                                               
      )                                                             
  ```                                                               
  `unresolved -> auto_remediating` is already an allowed transition. The `xfail` marker in `test_negative_balance_customized_spend_is_compared_with_real_decoder` (line 267) then goes away and the assertions it guards become live. One thing to decide: an issue that fails
  every day (receipt never available) grows its attempts list by one entry per day, so either dedupe consecutive identical failures or only retry when the last failure is older than the refresh period.

File: rotkehlchen/chain/evm/decoding/decoder.py [hunk 7]
```diff
@@ -817,6 +834,31 @@ def _decode_transaction(
...
+
+        The regular decoder's deferred-write path is used so sequence-index normalization is
+        identical to a persisted decode. The buffer is deliberately discarded, leaving both the
+        history events and the transaction's decoded marker unchanged.
+
+        Rule failures propagate instead of returning partially decoded events.
+        """
+        with self.undecoded_tx_query_lock:
+            with self.database.conn.read_ctx() as cursor:
+                self.reload_data(cursor)
```
Comment (LefterisJP) [inline rotkehlchen/chain/evm/decoding/decoder.py]: This reloads on every preview, while `_do_decode_transaction_hashes` reloads once per batch. On EVM `_reload_single_decoder` runs `ReloadableCacheDecoderMixin.reload_data()` for every cache-backed decoder (curve, velodrome, pendle, morpho, beefy, yearn, stakedao, ...): a
  freshness check, a possible remote refresh, and a full re-read of the global DB cache tables into new dicts. On the empty test DB that is 2.2 ms across 119 decoders; on a populated global DB it is considerably more, multiplied by the number of customized transactions of
  the issue.                                                        

  Suggest reloading once per chain per run instead, e.g. a `reload: bool = True` parameter here:

  ```suggestion
              if reload:                                            
                  with self.database.conn.read_ctx() as cursor:     
                      self.reload_data(cursor)
  ```
  and in `run_data_issue_remediation` keep a `reloaded_chains: set[ChainID]` next to `preview_cache`, passing `reload=chain_id not in reloaded_chains` from `_preview_transaction` and adding the chain afterwards. The lock is still taken per preview, which is fine.

File: rotkehlchen/tasks/manager.py [hunk 8]
```diff
@@ -680,6 +687,39 @@ def _maybe_process_historical_balances(self) -> list[Task] | None:
...
+            ) is None:
+                return None
+
+        if not force and should_run_periodic_task(
+            database=self.database,
+            key_name=DBCacheStatic.LAST_DATA_ISSUE_REMEDIATION_TS,
+            refresh_period=DATA_ISSUE_REMEDIATION_REFRESH,
+            cached_timestamps=self._scheduler_task_timestamps,
+        ) is False:
+            return None
```
Comment (LefterisJP) [inline rotkehlchen/tasks/manager.py]: The `LAST_HISTORICAL_BALANCE_PROCESSING_TS` read hits the DB on every 10 s scheduler tick, before the cooldown check that is served from `_scheduler_task_timestamps`. Checking the cooldown first means the DB read only happens once a day (or on a forced trigger):

  ```suggestion
          if not force and should_run_periodic_task(                
              database=self.database,                               
              key_name=DBCacheStatic.LAST_DATA_ISSUE_REMEDIATION_TS,
              refresh_period=DATA_ISSUE_REMEDIATION_REFRESH,        
              cached_timestamps=self._scheduler_task_timestamps,
          ) is False:
              return None                                           

          with self.database.conn.read_ctx() as cursor:             
              if self.database.get_static_cache(                    
                  cursor=cursor,
                  name=DBCacheStatic.LAST_HISTORICAL_BALANCE_PROCESSING_TS,
              ) is None:
                  return None                                       
  ```
  `test_data_issue_remediation_runs_daily_after_initial_processing` still passes with this order.

File: rotkehlchen/api/v1/types.py [hunk 9]
```diff
@@ -35,3 +35,5 @@ class TaskName(SerializableEnumNameMixin):
     HISTORICAL_BALANCE_PROCESSING = auto()
     ASSET_MOVEMENT_MATCHING = auto()
     BRIDGE_MATCHING = auto()
+    if __debug__:
+        DATA_ISSUE_REMEDIATION = auto()
```
Comment (LefterisJP) [inline rotkehlchen/api/v1/types.py]:
```suggestion
      DATA_ISSUE_REMEDIATION = auto()                               
  ```                                                               
  An enum member that only exists under `__debug__` is surprising to read and forces the two-subprocess test in `test_data_issues.py` (`test_remediation_task_api_validation_requires_debug`, from line 56), each of which imports all of rotki. Defining the member
  unconditionally and gating in the handler gives the same release behaviour with less machinery:                                       

  ```python                                                         
          elif task == TaskName.DATA_ISSUE_REMEDIATION:             
              if not __debug__ or is_accounting_update_enabled() is False:                                                              
                  return wrap_in_fail_result(
                      message='Data issue remediation is disabled',
                      status_code=HTTPStatus.NOT_FOUND,             
                  )
  ```                                                               
  Then the subprocess test can go, and `test_trigger_data_issue_remediation` covers the rest. Not a bug, a simplification.

**Review by LefterisJP — COMMENTED** [review]
Some scope questions, since this differs from what #12289 describes and the issue text was not updated after the re-scoping comment:

  1. There is no `BaseRemediationStrategy` / `RemediationPipeline` / per-strategy time budget; it is one hard-coded strategy in `tasks/data_issues.py`. #12290 (refetch strategy) will need a place to plug in. Is dropping the framework intentional, and if so should the issue be edited to match?
  2. No WebSocket message is sent (the issue asked for `PROGRESS_UPDATES` or a dedicated subtype). The frontend only learns about an attempt on refetch. Deferred to the UI PR on purpose?
  3. `customized_transaction_count` counts every customized transaction of the account before `ts_end`, regardless of asset, so the UI will report N transactions even when most never touch the bucket. Should it count only transactions whose saved or previewed events touch 
  the bucket?
  4. The preview still does network I/O (token lookups via `get_or_create_evm_token`, protocol cache refresh in `reload_data`). The issue framed this strategy as having no external dependencies.

**Comment by yabirgb** [comment]
> There is no BaseRemediationStrategy / RemediationPipeline / per-strategy time budget; it is one hard-coded strategy in tasks/data_issues.py. https://github.com/rotki/rotki/issues/12290 (refetch strategy) will need a place to plug in. Is dropping the framework intentional, and if so should the issue be edited to match?

I'll see about exactly defining the framework when introducing more tasks that run and send messages to the frontend

> No WebSocket message is sent (the issue asked for PROGRESS_UPDATES or a dedicated subtype). The frontend only learns about an attempt on refetch. Deferred to the UI PR on purpose?

Yes, it is in the follow up PR

> customized_transaction_count counts every customized transaction of the account before ts_end, regardless of asset, so the UI will report N transactions even when most never touch the bucket. Should it count only transactions whose saved or previewed events touch
the bucket?

The changes made do that

> The preview still does network I/O (token lookups via get_or_create_evm_token, protocol cache refresh in reload_data). The issue framed this strategy as having no external dependencies.

token assets should already exist because the transaction was decoded previously. Protocol cache reloads can still potentially refresh remotely, but changes limits them to once per chain per remediation run.

File: rotkehlchen/tasks/data_issues.py [hunk 10]
```diff
@@ -0,0 +1,290 @@
...
+                changed_transaction_count += 1
+            checkpoint()
+    except preview_exceptions as e:
+        log.exception('Failed to preview customized transactions for data issue %s', issue.id)
+        attempt = _make_comparison_attempt(
+            result='redecoding_failed',
+            customized_transaction_count=customized_transaction_count,
+            changed_transaction_count=changed_transaction_count,
+            reason=str(e),
+        )
```
Comment (LefterisJP) [inline rotkehlchen/tasks/data_issues.py:232]: A cancellation strands the issue in `auto_remediating`. `TaskCancelledError` is a `BaseException`, so a logout while a preview runs escapes this handler after `update_state(AUTO_REMEDIATING)` on line 187 already committed. The daily loop then never revisits it: line 261
  only accepts `open` issues or ones whose last attempt failed, and `retry_auto_remediation` rejects negative-balance issues, so the only way out for the user is to dismiss it. Recording the cancellation as a failed attempt before re-raising lets the existing retry path   
  pick it up on the next run.                                       

  ```suggestion                                                     
      except preview_exceptions as e:
          log.exception('Failed to preview customized transactions for data issue %s', issue.id)
          attempt = _make_comparison_attempt(
              result='redecoding_failed',
              customized_transaction_count=customized_transaction_count,
              changed_transaction_count=changed_transaction_count,
              reason=str(e),                                        
          )
      except TaskCancelledError:
          issues_manager.update_state(
              issue_id=issue.id,                                                                                                        
              state=IssueState.UNRESOLVED,
              attempt=_make_comparison_attempt(                                                                                                                                                                                                                                  
                  result='redecoding_failed',
                  customized_transaction_count=customized_transaction_count,
                  changed_transaction_count=changed_transaction_count,
                  reason='Remediation was cancelled before the comparison finished',                                         
              ),
          )                                                                                                                                                                                                                                                                      
          raise                                                             
  ```
     
  `TaskCancelledError` is exported from `rotkehlchen.concurrency`, so line 5 becomes `from rotkehlchen.concurrency import TaskCancelledError, checkpoint`. A test for it in `test_data_issue_remediation.py`: patch `_preview_transaction` with
  `side_effect=TaskCancelledError()`, assert `run_data_issue_remediation` raises, the issue is `unresolved` with a `redecoding_failed` attempt, and a second run with a normal preview calls the preview again and records the comparison

File: rotkehlchen/tasks/data_issues.py [hunk 11]
```diff
@@ -0,0 +1,290 @@
...
+            ),
+            customized_transaction_count=customized_transaction_count,
+            changed_transaction_count=changed_transaction_count,
+        )
+
+    issues_manager.update_state(
+        issue_id=issue.id,
+        state=IssueState.UNRESOLVED,
+        attempt=attempt,
+    )
```
Comment (LefterisJP) [inline rotkehlchen/tasks/data_issues.py:260]: The retry-on-failure loop has no dedupe, so an issue whose receipt never becomes available (or whose token lookup keeps failing) gains one identical `redecoding_failed` entry per day for as long as the user keeps rotki open. This keeps the first entry when the new
  failure is identical to the previous one; the state transition still happens so the run is accounted for.                             
                                                                                                                                                                                                                                                                                 
  ```suggestion                                                                                                                                                                                                                                                                  
      issues_manager.update_state(                                  
          issue_id=issue.id,                                        
          state=IssueState.UNRESOLVED,
          attempt=None if _is_repeated_failure(issue, attempt) else attempt,                                                            
      )
  ```                                                                                                                                                                                                                                                                            

  with, next to `_last_attempt_failed`:                                                                                                 
                                                                                                                                        
  ```python                                                                                                                    
  def _is_repeated_failure(issue: DataIssue, attempt: AutoRemediationAttempt) -> bool:
      """A failure identical to the previous attempt keeps the earlier timeline entry."""                                               
      return (
          attempt.get('result') == 'redecoding_failed' and          
          len(issue.auto_remediation_attempts) != 0 and             
          {k: v for k, v in issue.auto_remediation_attempts[-1].items() if k != 'timestamp'} ==                                                                                                                                                                                  
          {k: v for k, v in attempt.items() if k != 'timestamp'}    
      )                                                                                                                                                                                                                                                                          
  ```                                                                                                                                                                                                                                                                            

  The `missing_receipt` case of `test_negative_balance_customized_spend_is_compared_with_real_decoder` still expects two attempts, since the second one succeeds. Worth adding a case where the receipt stays missing across two runs and asserting a single attempt. If you
  would rather show the latest timestamp instead of the first, the manager needs a replace variant of `append_auto_remediation_attempt`; either is fine by me.

File: rotkehlchen/tasks/data_issues.py [hunk 12]
```diff
@@ -0,0 +1,290 @@
...
+    Saved events are never removed or replaced. Each applicable issue receives a diagnostic
+    timeline entry and remains unresolved for the user to review.
+    """
+    issues_manager = DataIssuesManager(database)
+    treat_eth2_as_eth = CachedSettings().get_entry('treat_eth2_as_eth') is True
+    preview_cache: PreviewCache = {}
+    reloaded_chains: set[EVM_CHAIN_IDS_WITH_TRANSACTIONS_TYPE] = set()
+    for issue in issues_manager.list_issues(DataIssueFilters(kind=IssueKind.NEGATIVE_BALANCE)):
+        if issue.state != IssueState.OPEN and _last_attempt_failed(issue) is False:
+            continue
```
Comment (LefterisJP) [inline rotkehlchen/tasks/data_issues.py]: Dropping the state filter to make the retry work means every negative-balance issue is now loaded daily, resolved and dismissed ones included, each with its payload and attempts JSON deserialized. That set only grows over the life of a DB. `DataIssuesFilterQuery` already
  supports a list of states, so the query can stay bounded:         

  ```suggestion                                                     
      for issue in issues_manager.list_issues(DataIssuesFilterQuery.make(
          kinds=[IssueKind.NEGATIVE_BALANCE],
          states=[IssueState.OPEN, IssueState.UNRESOLVED],          
      )):
          if issue.state != IssueState.OPEN and _last_attempt_failed(issue) is False:                                                   
              continue                                              
  ```

  `DataIssuesFilterQuery` comes from `rotkehlchen.db.filtering` (already imported from there), and `DataIssueFilters` becomes unused in this module. If you take the cancellation fix above, `auto_remediating` does not need to be in the list, since a cancelled issue is moved
  to `unresolved` before the task exits.

File: rotkehlchen/tasks/data_issues.py [hunk 13]
```diff
@@ -0,0 +1,290 @@
...
+            ),
+        )]
+        if len(customized_group_identifiers) == 0:
+            return {}
+
+        transaction_events = dbevents.get_history_events_internal(
+            cursor=cursor,
+            filter_query=EvmEventFilterQuery.make(
+                location=location,
+                group_identifiers=list(customized_group_identifiers),
```
Comment (LefterisJP) [inline rotkehlchen/tasks/data_issues.py]:
```suggestion
                  group_identifiers=customized_group_identifiers,
  ```
  Already a list, isn't it?

File: rotkehlchen/tasks/data_issues.py [hunk 14]
```diff
@@ -0,0 +1,290 @@
...
+                preview_cache[chain_id, tx_hash] = preview_events
+            preview_effects = _get_bucket_effects(preview_events, bucket, treat_eth2_as_eth)
+            if len(saved_effects) == 0 and len(preview_effects) == 0:
+                continue
+
+            if len(saved_effects) == 0:
+                customized_transaction_count += 1
+            if saved_effects != preview_effects:
+                changed_transaction_count += 1
+            checkpoint()
```
Comment (LefterisJP) [inline rotkehlchen/tasks/data_issues.py]: The `continue` on line 215 skips this checkpoint, so a long run of customized transactions that do not touch the bucket never checks for cancellation between previews. Moving `checkpoint()` to the top of the loop body (before line 200) covers every iteration

File: rotkehlchen/api/rest.py [hunk 15]
```diff
@@ -3872,7 +3872,21 @@ def trigger_task(self, task: TaskName) -> dict[str, Any]:
...
-
+        elif task == TaskName.DATA_ISSUE_REMEDIATION:
+            if not __debug__ or is_accounting_update_enabled() is False:
+                return wrap_in_fail_result(
+                    message='Data issue remediation is disabled',
+                    status_code=HTTPStatus.NOT_FOUND,
+                )
+            if (
+                (task_manager := self.rotkehlchen.task_manager) is None or
+                task_manager._maybe_run_data_issue_remediation(force=True) is None
```
Comment (LefterisJP) [inline rotkehlchen/api/rest.py]: Nit: this reaches into a private scheduler method from the API layer, where the historical-balance trigger goes through the public `trigger_historical_balance_processing`. A small public `trigger_data_issue_remediation(self) -> bool` on `TaskManager` that wraps the      
  `force=True` call would keep the boundary the same for both.

**Review by LefterisJP — COMMENTED** [review]
Left some comments

**Review by LefterisJP — APPROVED** [review]
LGTM
